2025-06-27 20:50:23,561 - INFO - Training with config: {'dataset_name': 'tinystory', 'context_length': 256, 'batch_size': 8, 'device': 'cpu', 'vocab_size': 10000, 'context_size': 1024, 'num_layers': 4, 'd_model': 512, 'num_heads': 16, 'd_ff': 1344, 'attn_pdrop': 0.1, 'resid_pdrop': 0.1, 'init_from': 'scratch', 'total_iters': 20000, 'warmup_iters': 200, 'lr_max': 0.0005, 'lr_min': 0, 'weight_decay': 0.001, 'wandb_logging': True, 'wandb_project': 'cs336-assignment1', 'wandb_run_name': 'tinystories-baseline', 'log_interval': 20, 'eval_interval': 200, 'eval_iters': 100, 'no_rmsnorm': False, 'parallel_layers': False, 'post_norm': False}
Traceback (most recent call last):
  File "/Users/charon/miniconda3/envs/llm/lib/python3.10/runpy.py", line 187, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/Users/charon/miniconda3/envs/llm/lib/python3.10/runpy.py", line 110, in _get_module_details
    __import__(pkg_name)
  File "/Users/charon/Documents/GitHub/CS336/basics/train.py", line 77, in <module>
    dataset = Dataset(**asdict(config))
  File "/Users/charon/Documents/GitHub/CS336/basics/data.py", line 37, in __init__
    self.val_data = np.memmap(f'{dataset_path}/val.bin', dtype=np.uint16, mode='r').astype(np.int64)
  File "/Users/charon/miniconda3/envs/llm/lib/python3.10/site-packages/numpy/_core/memmap.py", line 234, in __new__
    f_ctx = open(
FileNotFoundError: [Errno 2] No such file or directory: 'data/tinystory/val.bin'
Traceback (most recent call last):
  File "/Users/charon/miniconda3/envs/llm/lib/python3.10/runpy.py", line 187, in _run_module_as_main
    mod_name, mod_spec, code = _get_module_details(mod_name, _Error)
  File "/Users/charon/miniconda3/envs/llm/lib/python3.10/runpy.py", line 110, in _get_module_details
    __import__(pkg_name)
  File "/Users/charon/Documents/GitHub/CS336/basics/train.py", line 77, in <module>
    dataset = Dataset(**asdict(config))
  File "/Users/charon/Documents/GitHub/CS336/basics/data.py", line 37, in __init__
    self.val_data = np.memmap(f'{dataset_path}/val.bin', dtype=np.uint16, mode='r').astype(np.int64)
  File "/Users/charon/miniconda3/envs/llm/lib/python3.10/site-packages/numpy/_core/memmap.py", line 234, in __new__
    f_ctx = open(
FileNotFoundError: [Errno 2] No such file or directory: 'data/tinystory/val.bin'
